src/mlpack/core/math/shapiro_wilk_test.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file shapiro_wilk_test.hpp
+ *
+ * Implementation of Exteneded Shapiro Wilk Test for normality
+ * Refrences: https://tinyurl.com/q9gx9m2
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_MATH_SHAPIRO_WILK_TEST_IMPL_HPP
+#define MLPACK_CORE_MATH_SHAPIRO_WILK_TEST_IMPL_HPP
+
+
+#include <mlpack/prereqs.hpp>
+#include <boost/math/distributions.hpp>
+#include <boost/math/distributions/normal.hpp>
+
+namespace mlpack {
+namespace math {
+
+struct Pval
+{
+  double pvalue;
+  double W;
+  double zscore;
+  bool accept;
+};
+
+inline double InverseNormal(double prob, double mean =0 , double sd =1)
+{
+        boost::math::normal_distribution<>myNormal (mean, sd);
+        return quantile(myNormal, prob);
+};
+template<typename eT>
+Pval Shapiro(arma::Mat<eT>& dist, double alpha=0.05)
+{
+  //All the variable are consitent with refrence provided above
+  std::mt19937 engine;  // Mersenne twister random number engine
+  std::normal_distribution<> distr(0, 1.0);
+  dist.reshape(dist.n_rows*dist.n_cols, 1);
```

Comment:
Please don't use your own RNG, you can use `math::Random()` and other similar functionality.  The reason for this is that `math::RandomSeed()` sets the seed for all RNGs that we use, which makes it way easier to debug tests.